src/scaler/spreadscaler.rs

```diff
@@ -0,0 +1,1197 @@
+                            host_id: first_host.id.to_owned(),
+                            count: count - current_count,
+                            model_name: self.config.model_name.to_owned(),
+                            annotations: self.annotations(&spread.name),
+                        })),
+                        // Stop actors to reach desired replicas
+                        Ordering::Greater => Some(Command::StopActor(StopActor {
+                            actor_id: actor_id.to_owned(),
+                            host_id: first_host.id.to_owned(),
+                            count: current_count - count,
+                            model_name: self.config.model_name.to_owned(),
+                            annotations: self.annotations(&spread.name),
+                        })),
+                    }
+                } else {
+                    // No hosts were eligible, so we can't attempt to add or remove actors
+                    // We will want to return an error to indicate an unsatisfied Scaler here, eventually
+                    None
+                }
+            })
+            .collect::<Vec<Command>>();
+
+        Ok(commands)
+    }
+}
+
+impl<S: ReadStore + Send + Sync> ActorSpreadScaler<S> {
+    #[allow(unused)]
+    /// Construct a new ActorSpreadScaler with specified configuration values
+    fn new(
+        store: S,
+        actor_reference: String,
+        lattice_id: String,
+        model_name: String,
+        spread_config: SpreadScalerProperty,
+    ) -> Self {
+        Self {
+            store,
+            config: ActorSpreadConfig {
+                actor_reference,
+                lattice_id,
+                spread_config,
+                model_name,
+            },
+        }
+    }
+
+    // TODO(brooksmtownsend): Need to consider a better way to grab this. It will fail to find
+    // the actor ID if it's not running in the lattice currently, which seems brittle
+    /// Helper function to retrieve the actor ID for the configured actor
+    async fn actor_id(&self) -> Result<String> {
+        Ok(self
+            .store
+            .list::<Actor>(&self.config.lattice_id)
+            .await?
+            .iter()
+            .find(|(_id, actor)| actor.reference == self.config.actor_reference)
+            .map(|(id, _actor)| id.to_owned())
+            // Default here means the below `get` will find zero running actors, which is fine because
+            // that accurately describes the current lattice having zero instances.
+            .unwrap_or_default())
+    }
+
+    /// Helper function to create a predictable annotations map for a spread
+    fn annotations(&self, spread_name: &str) -> HashMap<String, String> {
+        HashMap::from_iter([
+            (SCALER_KEY.to_string(), SCALER_VALUE.to_string()),
+            (SPREAD_KEY.to_string(), spread_name.to_string()),
+            (
+                crate::MANAGED_BY_ANNOTATION.to_owned(),
```

Comment:
These should just be handled by the command worker. It is probably better to modify that logic to merge in these values if they aren't set. I also think the scaler/model name should be on the commands and then get set automatically by the command handler to ensure we have a way to look up which scaler to use

Reply:
Update after reading further: Looks like you actually already did this, so these can be removed. My comment about having the command handler automatically label the name seems like a better idea for us so we can look things up

Reply:
I think the model name should be an externally set annotation since we'll use it to key on. The spread name is internal to the scaler implementation

Reply:
After talking to @brooksmtownsend the only thing we need to do here is remove the managed by annotation since the main wadm code takes care of it